app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MovementService.scala

```diff
@@ -63,6 +65,19 @@ class MovementService @Inject()(
     movementRepository.getMovementById(id)
   }
 
+  def validateMovementId(id: String): Future[Either[MovementValidationError, Movement]] = {
```

Comment:
I would rename this to getMovementId and call the movementRepository.getMovementById(id) here and delete the current getMovementId. I Know things function does two things (violate the single responsibility principle) but we will always validate the Id before getting the movementId from the repository so I think this is ok. There are few places where we validate the Id and then get the movementId from db.

Reply:
I've moved this validateMovementId to its own class